extensions/amp-story/amp-story-ads.md

```diff
@@ -0,0 +1,176 @@
+    "impressionUrl": "https://adserver.com/track?iid=18745543"
+  },
+  "var": {
+    "ctaType": "EXPLORE",
+    "ctaUrl": "https://advertiser.com/landing-123.html",
+    "impressionId": "ac2d1s2E3B"
+  }
+}
+```
+
+* `templateId`: the ID of the inlined template that is going to be used.
+* `data`: the data model to populate the selected template. The fields should match the variable names in the selected template.
+* `var`: extra variables needed by the story. They will be added to the `amp-ad` element as data attributes, and picked by runtime for different use cases:
+       * CTA button rendering (see details in the ["CTA ad" section](#cta-ad))
+       * Provide dynamic content of the ad for tracking purpose (see details in the "tracking" section)
+
+#### Response headers
+The ad request will be an AMPCORS request, hence a couple of custom response headers are needed.
+See [AMPCORS spec](../../spec/amp-cors-requests.md).
+
+### Tagging
+
+In AMP Story, you cannot put `amp-ad` directly onto the page, instead, all ads
+are fetched & displayed by the [amp-story-auto-ads](./amp-story-auto-ads.md)
+extension.
+
+Here is a full example using `amp-story-auto-ads` together with some templates inlined.
+
+```html
+<amp-story>
+  <amp-story-auto-ads>
+     <script type=”application/json”>
+       {
+          "ad-attributes": {
+            type: “custom”
+            data-src: “https://adserver.com/getad?slot=abcd1234”
+          }
+       }
+     </script>
+
+     <template type="amp-mustache" id="template-1">
+       <amp-img src="{{imgSrc}}"></amp-img>
+       <amp-pixel src="{{impressionUrl}}"></amp-pixel>
+     </template>
+
+     <template type="amp-mustache" id="template-2">
+       <div class="creative-line-1">{{creativeLine1}}</div>
+       <div class="creative-line-2">{{creativeLine2}}</div>
+       <amp-pixel src="{{impressionUrl}}"></amp-pixel>
+     </template>
+  </amp-story-auto-ads>
+  ...
+```
+
+At runtime, an `amp-ad` element will be inserted dynamically:
+
+```html
+<amp-ad type="custom"
+  data-src="https://adserver.com/getad?slot=abcd1234"
+</amp-ad>
+```
+
+And an ad request will be made to this URL `https://adserver.com/getad?slot=abcd1234`.
+Each story can only have one `amp-story-auto-ads` element.
+
+### CTA ad
+To provide a consistent user experience, the story runtime will be responsible to render 
+the button of a CTA ad. The URL and button text will be provided in the `var` 
+object of the ad response.
+
+*  `ctaType`: the CTA button type, of which the value is an enum 
+   * EXPLORE: "Explore Now"
+   * SHOP: "Shop Now"
+   * READ: "Read Now"
+   * If you need support for a new CTA button, please open a [GitHub issue](https://github.com/ampproject/amphtml/issues/new)
+* `ctaUrl`: the landing page URL for the CTA button
+
+### Tracking
+Each story page that is dynamically inserted for ad will be assigned with a system generated page ID, prefixed with `i-amphtml-AD-`. The `story-page-visible` trigger 
```

Comment:
do we want `i-amphtml-AD-` or i-amphtml-Ad-`. I don't feel strongly but I think the code has Ad for now